visualization/app/codeCharta/ui/searchPanel/mapTreeView/mapTreeViewItemIcon/mapTreeViewItemIconColor.pipe.ts

```diff
@@ -4,16 +4,23 @@ import { CodeMapNode } from "../../../../codeCharta.model"
 import { Store } from "../../../../state/store/store"
 import { getMarkingColor, isLeaf } from "../../../../util/codeMapHelper"
 
-@Pipe({ name: "mapTreeViewItemIconColor" })
+@Pipe({ name: "mapTreeViewItemIconColor", pure: false })
 export class MapTreeViewItemIconColorPipe implements PipeTransform {
 	static defaultColor = "#000000"
+	static areMetricZeroColor = "#BDBDBD"
 	static store = Store.store
 
 	transform(value: CodeMapNode): string | undefined {
-		if (isLeaf(value)) {
-			return undefined
+		const { areaMetric } = MapTreeViewItemIconColorPipe.store.getState().dynamicSettings
+
+		if (value.attributes[areaMetric]) {
```

Comment:
Just a suggestion for more readability (less indentation due to early returns).

```suggestion
		if (!value.attributes[areaMetric]) {
			return MapTreeViewItemIconColorPipe.areMetricZeroColor
		}
		// ... other codnditions ...
```